### PR TITLE
Switch fx_ver_t::as_str from stringstream to append/to_string

### PR DESCRIPTION
**Pull request by elinor-fung**
Stop using `std::stringstream` to convert `fx_ver_t` to a string. It pulls in `std::locale` (and other things) in all the host binaries when we really just need a simple conversion of the numbers to a string. In some binaries, it was also the only usage of stringstream. In theory, this should also be faster, although this is not a particularly perf-sensitive function (most usage is for when tracing is enabled).

On my local Windows x64 build, the sizes on disk were:
| | before (kB) | after (kB) | diff (kB) |
| --- | ---: | ---: | ---: |
| apphost | 153 | 136 | -17 |
| comhost | 189 |  174 | -15 |
| dotnet | 172 | 129 | -43 |
| ijwhost | 141 | 101 | -40 |
| nethost | 131 | 88 | - 43 |

### CODE REVIEW

File: src/native/corehost/fxr/fx_ver.cpp [hunk 1]
```diff
@@ -70,31 +70,18 @@ bool fx_ver_t::operator >=(const fx_ver_t& b) const
 
 pal::string_t fx_ver_t::as_str() const
 {
-    pal::stringstream_t stream;
-    stream << m_major << _X(".") << m_minor << _X(".") << m_patch;
+    pal::string_t version = pal::to_string(m_major);
```
Comment (am11) [inline src/native/corehost/fxr/fx_ver.cpp:73]: Can we remove  `#include <sstream>` and `#include <locale>` from pal.* files?

**Review by vitek-karas — APPROVED** [review]
Nice 👍

**Review by jkoritzinsky — APPROVED** [review]

File: src/native/corehost/fxr/fx_ver.cpp (hunk 1 above)
Comment (elinor-fung) [reply src/native/corehost/fxr/fx_ver.cpp:73]: Some things still use stringstream, so we still need sstream. I think locale can go though.

**Pull request merged**